HouseRobber

```diff
@@ -1,16 +1,28 @@
 class Solution {
     public int rob(int[] nums) {
         int n = nums.length;
-        int evenSum =0, oddSum =0;
-        for(int i=0;i<n;i=i+2){
-            evenSum += nums[i];
+        if(nums.length==0 || nums == null){
+            return 0;
         }
-        for(int j=1;j<n;j=j+2){
-            oddSum += nums[j];
+        int[][] dp = new int[n][2];
+        
+        dp[0][0] = nums[0];
+        dp[0][1] = 0;
+        for(int i=1;i<n;i++){
+            for(int j=0;j<2;j++){
+                if(j==0){
+                    dp[i][0] = dp[i-1][1] + nums[i];
+                }
+                else{
+                    dp[i][1] = Math.max(dp[i-1][0],dp[i-1][1]); 
+                }
+            }
         }
-        int max = Math.max(evenSum,oddSum);
-        return max;
+        return Math.max(dp[n-1][0],dp[n-1][1]);
     }
 }
 
-//39/69 testcases passed! Not getting exact idea..
+//Time Complexity: O(n * 2) = O(2) // Order of n for i and 2 for j
```

Comment:
Good Job! But small error over here. Did you mean O(n)? You can contact me, the cohort TA or your professor for any help.